pixeltable/catalog/catalog.py

```diff
@@ -103,7 +103,14 @@ def loop(*args: Any, **kwargs: Any) -> T:
                     Catalog.get()._finalize_pending_ops(e.tbl_id)
                 except (sql_exc.DBAPIError, sql_exc.OperationalError) as e:
                     # TODO: what other exceptions should we be looking for?
-                    if isinstance(e.orig, (psycopg.errors.SerializationFailure, psycopg.errors.LockNotAvailable)):
+                    if isinstance(
+                        e.orig,
+                        (
+                            psycopg.errors.SerializationFailure,
+                            psycopg.errors.LockNotAvailable,
+                            psycopg.errors.DeadlockDetected,
```

Comment:
Leave todo: investigate whether DeadlockDetected points to a bug in our locking protocol, which is supposed to be deadlock-free.